props/saber_fett263_buttons.h

```diff
@@ -285,47 +440,1090 @@ EFFECT(faston); // for EFFECT_FAST_ON
+    }
+    saved_color_ = GetColorArg(blade_num_, effect_num_);
+    hsl = saved_color_.toHSL();
+    NewColor(blade_num_, effect_num_);
+    current_preset_.Save();
+    effect_num_ = 0;
+  }
+
+  void SetInOut() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    switch (menu_type_) {
+      case MENU_STYLE_OPTION:
+      case MENU_IGNITION_TIME:
+      case MENU_IGNITION_OPTION:
+      case MENU_PREON_OPTION:
+      case MENU_PREON_SIZE:
+        UpdateFont(current_preset_.preset_num);
+        break;
+      case MENU_RETRACTION_TIME:
+      case MENU_RETRACTION_OPTION:
+        UpdateStyle(current_preset_.preset_num);
+        off_event_ = true;
+        last_rotate_ = millis();
+        break;
+      default:
+        break;
+    }
+  }
+
+  void RevertInOut() {
+    char style_arg[10];
+    itoa(arg_revert_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    switch (menu_type_) {
+      case MENU_RETRACTION_OPTION:
+      case MENU_RETRACTION_TIME:
+        char ig[10];
+        itoa(ignite_time_, ig, 10);
+        current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, ig));
+        break;
+      default:
+        break;
+    }
+    current_preset_.Save();
+    UpdateFont(current_preset_.preset_num);
+  }
+
+  void SaveStyleSetting() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    PlayMenuSound("msave.wav");
+    switch (set_num_) {
+      case 21:
+        show_clash_location_.Stop(blade_num_);
+        break;
+      case 22:
+        show_drag_size_.Stop(blade_num_);
+        break;
+      case 23:
+        show_emitter_size_.Stop(blade_num_);
+      default:
+        break;
+    }
+    UpdateStyle(current_preset_.preset_num);
+    menu_type_ = MENU_STYLE_SETTING_SUB;
+    announce_menu_ = true;
+  }
+
+  void CancelStyleSetting() {
+    PlayMenuSound("mcancel.wav");
+    switch (set_num_) {
+      case 21:
+        show_clash_location_.Stop(blade_num_);
+        break;
+      case 22:
+        show_drag_size_.Stop(blade_num_);
+        break;
+      case 23:
+        show_emitter_size_.Stop(blade_num_);
+        break;
+      default:
+        break;
+    }
+    UpdateStyle(current_preset_.preset_num);
+    menu_type_ = MENU_STYLE_SETTING_SUB;
+    announce_menu_ = true;
+  }
+#endif
+
+  void NewColor(int blade, int effect) {
+    char new_color[32];
+    switch (color_mode_) {
+      case COLOR_LIST:
+      case CC_COLOR_LIST:
+        itoa(Color16(color_list_[dial_]).r, new_color, 10);
```

Comment:
Don't duplicate the color-to-string code.
Have a Color16 variable before the switch and assign it from hsl or color_list_[dial].
Then do the color-to-string after the switch.

Reply:
updated